src/wtforms/fields/choices.py

```diff
@@ -1,16 +1,63 @@
-import itertools
+import warnings
+from dataclasses import dataclass
+from typing import Optional
 
 from wtforms import widgets
 from wtforms.fields.core import Field
 from wtforms.validators import ValidationError
 
 __all__ = (
     "SelectField",
+    "Choice",
     "SelectMultipleField",
     "RadioField",
 )
 
 
+@dataclass
+class Choice:
+    """
+    A dataclass that represents available choices for
+    :class:`RadioField`, :class:`SelectField` and :class:`SelectMultipleField`
+
+    :param value:
+        The value that will be send in the request.
+    :param label:
+        The label of the option.
+    :param render_kw:
+        A dict containing HTML attributes that will be rendered
+        with the option.
+    :param optgroup:
+        The `<optgroup>` HTML tag in which the option will be
+        rendered.
+    """
+
+    value: str
+    label: Optional[str] = None
+    render_kw: Optional[dict] = None
+    optgroup: Optional[str] = None
+    _selected: bool = False
+
+    @classmethod
+    def from_input(cls, input, optgroup=None):
+        if isinstance(input, Choice):
+            if optgroup:
+                input.optgroup = optgroup
+            return input
+
+        if isinstance(input, str):
+            return Choice(value=input, optgroup=optgroup)
+
+        if isinstance(input, tuple):
+            warnings.warn(
+                "Passing SelectField choices as tuples is deprecated and will be "
+                "removed in wtforms 3.3. Please use Choice instead.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
```

Comment:
Please reconsider this. It will be a significant amount of busy-work for larger projects to adapt to this, without any real benefit. Just generating the Choice instance from the tuple is cheap and doesn't seem to have any drawbacks.

![image](https://github.com/wtforms/wtforms/assets/179599/bb982c96-f99e-432f-9d20-f2a791d1f233)

Yeah... I really do not want to spend time updating those almost 50 places in my app.